go/ql/lib/semmle/go/dataflow/internal/DataFlowPrivate.qll

```diff
@@ -310,6 +310,16 @@ class DataFlowCallable extends TDataFlowCallable {
     this.asSummarizedCallable()
         .hasLocationInfo(filepath, startline, startcolumn, endline, endcolumn)
   }
+
+  /** Gets a best-effort total ordering. */
+  int totalorder() {
+    this =
+      rank[result](DataFlowCallable c, string file, int startline, int startcolumn |
+        c.hasLocationInfo(file, startline, startcolumn, _, _)
```

Comment:
Why does the total order for `DataFlowCallable` take the file into account, but not those for `DataFlowCall` or `NodeRegion`?

Reply:
That's because we're ordering calls and regions within a specific callable body, so the ones we need to compare are always in the same file. But the callables that we need to compare can be spread across multiple files, so they benefit from the additional comparison column.
My hope is that this ordering is fairly temporary, as I'd like to get the MakeSets module turned into a QL primitive - once it's implemented in the evaluator then there's no need for these arbitrary orderings.

Reply:
That makes sense. In the cases where we are only comparing within the same file, would it make sense to limit the ordering to other instances in the same file? Or do you really want to have a total ordering over all calls, say, even if we only compare ones in the same file?

Reply:
A best-effort total ordering is simpler than trying to make it specifically per file - in the end it doesn't matter much, we just need some completely arbitrary ordering, and even in the cases where we fail to get this, the MakeSets module provides a fallback solution.